hoot-core/src/main/cpp/hoot/core/io/OsmApiChangeset.cpp

```diff
@@ -945,4 +1001,30 @@ void XmlChangeset::updateElement(ChangesetTypeMap& map, long old_id, long new_id
   }
 }
 
+bool XmlChangeset::fixElement(ChangesetTypeMap& map, long id, long version)
+{
+  bool success = false;
+  //  Negative IDs should never be fixed
+  if (id < 1)
+    return success;
+  //  Find the affected element
+  for (int type = 0; type < ChangesetType::TypeMax; ++type)
+  {
+    XmlElementMap::iterator position = map[type].find(id);
+    if (position != map[type].end())
+    {
+      //  Found the element, now update it
+      XmlElementPtr element = map[type][id];
+      //  Only update the version if it is out of sync
+      if (element->getVersion() != version)
+      {
+        element->setVersion(version);
+        //  Change was made
+        success = true;
+      }
+    }
+  }
+  return success;
+}
+
```

Comment:
Pretty simple, check the type map (map is either a map of nodes, ways, or relations) for the element and updated it if the versions are different.